swift/WorkflowUI/Sources/Container/WorkflowView.swift

```diff
@@ -0,0 +1,203 @@
+@available(iOS 13.0, *)
+fileprivate struct IntermediateView<T: Workflow, Content: View>: UIViewControllerRepresentable {
+    
+    var workflow: T
+    var onOutput: (T.Output) -> Void
+    var content: (T.Rendering) -> Content
+    
+    func makeUIViewController(context: UIViewControllerRepresentableContext<IntermediateView<T, Content>>) -> WorkflowHostingViewController<T, Content> {
+        WorkflowHostingViewController(workflow: workflow, content: content)
+    }
+    
+    func updateUIViewController(_ uiViewController: WorkflowHostingViewController<T, Content>, context: UIViewControllerRepresentableContext<IntermediateView<T, Content>>) {
+        uiViewController.content = content
+        uiViewController.onOutput = onOutput
+        uiViewController.update(to: workflow)
+    }
+    
+}
+
+@available(iOS 13.0, *)
+fileprivate final class WorkflowHostingViewController<T: Workflow, Content: View>: UIViewController {
+    
+    private let workflowHost: WorkflowHost<T>
+    private let hostingController: UIHostingController<RootView<Content>>
+    private let rootViewProvider: RootViewProvider<Content>
+    
+    var content: (T.Rendering) -> Content
+    var onOutput: (T.Output) -> Void
+    
+    private let (lifetime, token) = Lifetime.make()
+    
+    init(workflow: T, content: @escaping (T.Rendering) -> Content) {
+        
+        self.content = content
+        self.onOutput = { _ in }
+        
+        workflowHost = WorkflowHost(workflow: workflow)
+        rootViewProvider = RootViewProvider(view: content(workflowHost.rendering.value))
+        hostingController = UIHostingController(rootView: RootView(provider: rootViewProvider))
+        
+        super.init(nibName: nil, bundle: nil)
+        
+        addChild(hostingController)
+        view.addSubview(hostingController.view)
+        hostingController.didMove(toParent: self)
+        
+        workflowHost
+            .rendering
+            .signal
+            .take(during: lifetime)
+            .observeValues { [weak self] rendering in
+                self?.didEmit(rendering: rendering)
+            }
+        
+        workflowHost
+            .output
+            .take(during: lifetime)
+            .observeValues { [weak self] output in
+                self?.didEmit(output: output)
+            }
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        hostingController.view.frame = view.bounds
+    }
+    
+    private func didEmit(rendering: T.Rendering) {
+        rootViewProvider.view = content(rendering)
+    }
+    
+    private func didEmit(output: T.Output) {
+        onOutput(output)
+    }
+    
+    func update(to workflow: T) {
+        workflowHost.update(workflow: workflow)
+    }
+    
+}
```

Comment:
Should this plumb all the `childViewControllerFor…` methods through to `hostingController`? (Not exactly sure how status bar and whatnot interact with SwiftUI right now).

Reply:
They mostly... don't. But preferences go up the tree even if there are UIKit layers in the middle, so this should be ok.